File: alumnos/montactuaria/stdin.py
#!/usr/bin/env python

import re
import sys

while True:
  linea = sys.stdin.readline()

  if not linea:
    break

  linea = re.sub("minutes","m",linea,flags=re.IGNORECASE)
  linea = re.sub("minute","m",linea,flags=re.IGNORECASE)
  linea = re.sub("min","m",linea,flags=re.IGNORECASE)
  linea = re.sub("mi","m",linea,flags=re.IGNORECASE)
  linea = re.sub("seconds","s",linea,flags=re.IGNORECASE)
  linea = re.sub("second","s",linea,flags=re.IGNORECASE)
  linea = re.sub("sec","s",linea,flags=re.IGNORECASE)
  linea = re.sub("hours","h",linea,flags=re.IGNORECASE)
  linea = re.sub("hour","h",linea,flags=re.IGNORECASE)
  linea = re.sub("~", "", linea)
  linea = re.sub(r"[^msh0-9]", "", linea)
  print linea

File: alumnos/montactuaria/stdin.r
#!/usr/bin/env Rscript

f <- file("stdin")

open(f)

while(length(lines <- readLines(f, n = 1)) > 0) {
  lines <- gsub("minute|min|minutes","m",lines,ignore.case=TRUE)
  lines <- gsub("second|seconds|s", "s", lines,ignore.case=TRUE)
  lines <- gsub("hour|h|hours","h",lines,ignore.case=TRUE)
  lines <- gsub("[^msh0-9]","",lines)
  write(lines, stderr())
}

close(f)